N/NodeJS/NodeJS_14/build_tarballs.jl

```diff
@@ -0,0 +1,69 @@
+    ArchiveSource("$(url_prefix)-linux-arm64.tar.gz", "58cb307666ed4aa751757577a563b8a1e5d4ee73a9fac2b495e5c463682a07d1"; unpack_target = "aarch64-linux-gnu"),
+    ArchiveSource("$(url_prefix)-linux-ppc64le.tar.gz", "de6ccb9bf08520939cc2ae0507634015981604b5eb6912d031d4b7fe146f0de4"; unpack_target = "powerpc64le-linux-gnu"),
+    ArchiveSource("$(url_prefix)-linux-armv7l.tar.gz", "54efe997dbeff971b1e39c8eb910566ecb68cfd6140a6b5c738265d4b5842d24"; unpack_target = "arm-linux-gnueabihf"),
+
+    ArchiveSource("$(url_prefix)-linux-x64.tar.gz", "068400cb9f53d195444b9260fd106f7be83af62bb187932656b68166a2f87f44"; unpack_target = "x86_64-linux-musl"),
+    ArchiveSource("$(url_prefix)-linux-arm64.tar.gz", "58cb307666ed4aa751757577a563b8a1e5d4ee73a9fac2b495e5c463682a07d1"; unpack_target = "aarch64-linux-musl"),
+    ArchiveSource("$(url_prefix)-linux-armv7l.tar.gz", "54efe997dbeff971b1e39c8eb910566ecb68cfd6140a6b5c738265d4b5842d24"; unpack_target = "arm-linux-musleabihf"),
+
+    ArchiveSource("$(url_prefix)-darwin-x64.tar.gz", "b762b72fc149629b7e394ea9b75a093cad709a9f2f71480942945d8da0fc1218"; unpack_target = "x86_64-apple-darwin14"),
+    ArchiveSource("$(url_prefix)-darwin-x64.tar.gz", "b762b72fc149629b7e394ea9b75a093cad709a9f2f71480942945d8da0fc1218"; unpack_target = "aarch64-apple-darwin20"),
+
+    ArchiveSource("$(url_prefix)-win-x64.zip", "e469db37b4df74627842d809566c651042d86f0e6006688f0f5fe3532c6dfa41"; unpack_target = "x86_64-w64-mingw32"),
+    ArchiveSource("$(url_prefix)-win-x86.zip", "cfb3535a172fb792a63814deffde405466902359bedfbd884188f6fc56f97d64"; unpack_target = "i686-w64-mingw32"),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd ${WORKSPACE}/srcdir/
+cp -r ${target}/*/* ${prefix}/.
+cd ${prefix}
+if [[ "${target}" == *-mingw* ]]; then
+    mkdir bin
+    chmod +x {node.exe,npm,npm.cmd,npx,npx.cmd}
+    mv {node.exe,npm,npm.cmd,npx,npx.cmd} bin
+    mv node_modules bin
+fi
+install_license LICENSE
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = [
+    Platform("x86_64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("aarch64", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("powerpc64le", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+    Platform("armv7l", "linux"; libc="glibc", cxxstring_abi="cxx11"),
+
+    Platform("x86_64", "linux"; libc="musl", cxxstring_abi="cxx11"),
+    Platform("aarch64", "linux"; libc="musl", cxxstring_abi="cxx11"),
+    Platform("armv7l", "linux"; libc="musl", cxxstring_abi="cxx11"),
+
+    Platform("x86_64", "macos"),
+    Platform("aarch64", "macos"),
+
+    Platform("x86_64", "windows"),
+    Platform("i686", "windows"),
+]
+
+# The products that we will ensure are always built
+products = [
+    ExecutableProduct("node", :node),
+    FileProduct("bin/npm", :npm),
+    FileProduct("bin/npx", :npx),
```

Comment:
Actually, I'm just realizing that this won't work: `bin/npm` and `bin/npx` are cygwin specific (I think). Instead, on Windows there are `bin/npm.cmd` and `bin/npx.cmd` batch files present that we should actually expose. But not sure how we can do that, i.e. use a different path for Windows here?

Reply:
Could this be the pattern we are looking for? https://github.com/JuliaPackaging/Yggdrasil/blob/8be19820afa24ff817185f3c53c4d20786a8add3/A/AmazonCACerts/build_tarballs.jl#L46

Probably need to change the node.js 16 code as well, once we figure this out

Reply:
Looking into it further, less convinced the above link helps

Reply:
I seem to recall https://github.com/JuliaPackaging/Yggdrasil/blob/master/fancy_toys.jl#L15 being useful for dealing with differences in platforms.

Reply:
@jeremiahpslewis btw I've given you write access to my fork so you can just push direct to this branch to make this easier to finish off.

Reply:
If you don't want `bin/npm` and `bin/npx` at all in the Windows tarballs you can't delete them and have a `FileProduct` with multiple names.  But note that `FileProduct`s don't provide much convenience besides a variable pointing to their path, it's not like an `ExecutableProduct` for which we generate wrapper functions, so you may just ignore these two products in the JLL package.

Reply:
I think what we want is a `FileProduct` named `npm` that points to `bin/npm.cmd` on Windows, but to `bin/npm` everywhere else. Same for `npx`.

It would actually be nice if one could use `ExecutableProudct` for batch files...